server/services/withings/lib/withings.poll.js

```diff
@@ -0,0 +1,209 @@
+            // Convert type to int wihings
+            // (cf https://developer.withings.com/oauth2/#tag/measure%2Fpaths%2Fhttps%3A~1~1wbsapi.withings.net~1measure%3Faction%3Dgetmeas%2Fget)
+            let withingsType;
+            let featureBattery;
+            switch (feature.type) {
+              case DEVICE_FEATURE_TYPES.HEALTH.WEIGHT:
+                withingsType = 1;
+                break;
+              case DEVICE_FEATURE_TYPES.HEALTH.HEIGHT:
+                withingsType = 4;
+                break;
+              case DEVICE_FEATURE_TYPES.HEALTH.FAT_FREE_MASS:
+                withingsType = 5;
+                break;
+              case DEVICE_FEATURE_TYPES.HEALTH.FAT_RATIO:
+                withingsType = 6;
+                break;
+              case DEVICE_FEATURE_TYPES.HEALTH.FAT_MASS_WEIGHT:
+                withingsType = 8;
+                break;
+              case DEVICE_FEATURE_TYPES.HEALTH.DIASTOLIC_BLOOD_PRESSURE:
+                withingsType = 9;
+                break;
+              case DEVICE_FEATURE_TYPES.HEALTH.SYSTOLIC_BLOOD_PRESSURE:
+                withingsType = 10;
+                break;
+              case DEVICE_FEATURE_TYPES.HEALTH.HEARTH_PULSE:
+                withingsType = 11;
+                break;
+              case DEVICE_FEATURE_TYPES.HEALTH.TEMPERATURE:
+                withingsType = 12;
+                break;
+              case DEVICE_FEATURE_TYPES.HEALTH.SPO2:
+                withingsType = 54;
+                break;
+              case DEVICE_FEATURE_TYPES.HEALTH.BODY_TEMPERATURE:
+                withingsType = 71;
+                break;
+              case DEVICE_FEATURE_TYPES.HEALTH.SKIN_TEMPERATURE:
+                withingsType = 73;
+                break;
+              case DEVICE_FEATURE_TYPES.HEALTH.MUSCLE_MASS:
+                withingsType = 76;
+                break;
+              case DEVICE_FEATURE_TYPES.HEALTH.HYDRATION:
+                withingsType = 77;
+                break;
+              case DEVICE_FEATURE_TYPES.HEALTH.BONE_MASS:
+                withingsType = 88;
+                break;
+              case DEVICE_FEATURE_TYPES.HEALTH.PULSE_WAVE_VELOCITY:
+                withingsType = 91;
+                break;
+              case DEVICE_FEATURE_TYPES.SENSOR.INTEGER:
+                withingsType = -1;
+                featureBattery = { ...feature };
+                break;
+              default:
+                withingsType = 0;
+                break;
+            }
+
+            if (withingsType > 0) {
+              logger.debug('Current feature last value changed: ', feature.last_value_changed);
+
+              const measureResult = await oauth2Manager.executeOauth2HTTPQuery(
+                this.serviceId,
+                user.id,
+                'get',
+                `${this.withingsUrl}/measure`,
+                `action=getmeas&meastype=${withingsType}&category=1&lastupdate=${feature.last_value_changed.getTime() /
```

Comment:
Just to be sure I understand, where do you specify to Withings that you are querying this specific device?

Reply:
In action getmeas on withings API I can't  pass the deviceId, but I put withingsType (<=> Type of deviceFeature), this attribute limit the number of result downloaded.
I check the consistance between withingDeviceId (store in deviceParam) and devidId (from withings api) after api call.

So in fact if you have 2 device with same withingsType it is possible to download measure from 2 distinct device in 1 api request but save only data from current deviceFeature (and repeat this action for second device).